### ISSUE TITLE
Timex 2.2.1: ** (Timex.Parse.ParseError) Expected end of input for parsing dates using :strftime for certain timezones

### ISSUE THREAD

**Issue by marcelog** [bitwalker/timex#199]
Hello,

I'm getting the following error with Timex 2.2.1:
```elixir
Timex.parse! "2016-01-01T19:32:32 America/Port-au-Prince", "%Y-%m-%dT%H:%M:%S %Z", :strftime
** (Timex.Parse.ParseError) Expected end of input at line 1, column 32
    lib/parse/datetime/parser.ex:85: Timex.Parse.DateTime.Parser.parse!/3
```

As an example, the timezone America/Port-au-Prince exists as a valid Timezone when calling `Timex.timezones/0`.

The expected result would be to get a valid DateTime.

Perhaps I'm missing something?

Thanks in advance!

Update: this actually works:
```elixir
> now = DateTime.now
> now_in_tz = Timezone.convert now, "America/Port-au-Prince"
#<DateTime(2016-07-20T19:06:35 America/Port-au-Prince (-05:00:00))>
> Timex.format(now_in_tz, "%Y-%m-%dT%H:%M:%S %Z", :strftime)
{:ok, "2016-07-20T19:06:35 America/Port-au-Prince"}
```

**Comment by bitwalker** [comment]
Fixed in 3.0.5. I can potentially backport this fix if required, let me know.

**Comment by marcelog** [comment]
Thanks for the fix!

We've upgraded to 3.0.0 (and now 3.0.5) since then, so *we* don't need the backport (although perhaps other might benefit from it!)

Cheers!
